src/loadEnvironment.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from "path"
+import * as fs from "fs"
+import { ContentfulCollection, ContentTypeCollection, LocaleCollection } from "contentful"
+
+/* istanbul ignore next */
+const interopRequireDefault = (obj: any): { default: any } =>
+  obj && obj.__esModule ? obj : { default: obj }
+
+type Registerer = { enabled(value: boolean): void }
+
+let registerer: Registerer | null = null
+
+function enableTSNodeRegisterer() {
+  if (registerer) {
+    registerer.enabled(true)
+
+    return
+  }
+
+  try {
+    registerer = require("ts-node").register() as Registerer
```

Comment:
I'm struggling to find the documentation of `.register()` and `.enabled()`. Do you have that at hand? I'd love to learn more about this to be able to give you a better review

Reply:
You want to look [here for `register()`](https://typestrong.org/ts-node/api/index.html#register) and [here for `enabled()`](https://typestrong.org/ts-node/api/interfaces/Service.html#enabled) - that said they're just API docs so might not have the detail you want; I should be able to help answer any questions though 🙂

Reply:
The reason I've not used `Service` is because it lets you avoid importing from `ts-node`, which means there's less chance it can be "leaked" into the generated types (at which point `ts-node` would no longer be an optional peer dependency, as in some cases TypeScript can pickup that import even if you're not using a TypeScript config, and complain about the lack of `ts-node`).

Reply:
Makes sense! Let's leave it that way then 😄